plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -57,8 +58,28 @@ object ScalafmtPlugin extends AutoPlugin {
         "(By default this means the Compile and Test configurations.)"
     )
   }
+
+  val scalafmtCheckImpl =
+    taskKey[ScalafmtAnalysis](
+      "(Internal) Checks if a Scala source is mis-formatted. Does not write to files."
+    )
+  val scalafmtImpl =
+    taskKey[Long](
+      "(Internal) implements scalafmt."
+    )
+
   import autoImport._
 
+  case class ScalafmtAnalysis(failed: List[File], pending: List[File])
```

Comment:
is `pending` ever non-empty ?

Reply:
`checkSources` populates it as:

```scala
pending = (prev.pending ++ filesToCheck).distinct
```

so `scalafmt` can pick it up.

Reply:
I think I can get rid of pending.